Source/engine/render/text_render.cpp

```diff
@@ -92,14 +92,38 @@ text_color GetColorFromFlags(UiFlags flags)
 	return ColorWhitegold;
 }
 
-} // namespace
+std::array<uint8_t, 256> *LoadFontKerning(GameFontTables size, uint16_t row)
+{
+	uint32_t fontId = (size << 16) | row;
+
+	auto hotKerning = FontKerns.find(fontId);
+	if (hotKerning != FontKerns.end()) {
+		return hotKerning->second.get();
+	}
+
+	char path[32];
+	sprintf(path, "fonts\\%i-%02d.bin", FontSizes[size], row);
+
+	auto kerning = std::make_unique<std::array<uint8_t, 256>>();
+
+	LoadFileInMem(path, kerning.get());
+
+	return FontKerns.insert(make_pair(fontId, move(kerning))).first->second.get();

```

Comment:
After ditching the `unique_ptr`, you can do something like:
```
auto *kerning = &FontKerns[fontId];
LoadFileInMem(path, kerning);
return kerning;
```

Reply:
I need up to a 1024 arrays of 256 bytes depending on the language and chat content. But in most cases 4 arrays of 256 is all that is needed so having it in a variable size container allows me to save up to 0.2M of ram.

Reply:
Took me a few tries to figure out how this was suppose to work, but once i figured it out the code is much cleaner, so thanks :)